### ISSUE TITLE
3D dataset training

### ISSUE THREAD

**Issue by Tkrismile** [FengheTan9/Mobile-U-ViT#6]
I followed your README and used UNET's training and validation strategy. I first trained using main.py with the following parameters: 
python main.py --model_name MobileU --feature_size=32 --batch_size=1 --logdir=MobileU_test --optim_lr=1e-4 --lrschedule=warmup_cosine --infer_overlap=0.5 --save_checkpoint --data_dir=/home/ubuntu/data/BTCV
Then I tested using the test.py file with the following parameters:
python test.py --infer_overlap=0.5 --data_dir=/home/ubuntu/data/BTCV --pretrained_dir=‘./runs/MobileU_test’ --saved_checkpoint=ckpt
However, the results differ slightly from those in the paper. Why is this?

**Comment by Tkrismile** [comment]
Are the random seeds you used for training your 3D data the same as those used for training your 2D data?
seed_torch(41)

**Comment by Tkrismile** [comment]
I noticed that your paper's results on the BTCV dataset were obtained using five-fold cross-validation, but it appears that UNET was not trained using a five-fold approach. Could you provide your five-fold training code?Thank you very much.

**Comment by FengheTan9** [comment]
> I noticed that your paper's results on the BTCV dataset were obtained using five-fold cross-validation, but it appears that UNET was not trained using a five-fold approach. Could you provide your five-fold training code?Thank you very much.

Thank you for your attention to our work. By updating the JSON configuration (training and validataion list in dataset.json), the dataset is deterministically partitioned into five sequential folds for cross-validation, ensuring reproducibility.

**Comment by Tkrismile** [comment]
> > I noticed that your paper's results on the BTCV dataset were obtained using five-fold cross-validation, but it appears that UNET was not trained using a five-fold approach. Could you provide your five-fold training code?Thank you very much.
> 
> Thank you for your attention to our work. By updating the JSON configuration (training and validataion list in dataset.json), the dataset is deterministically partitioned into five sequential folds for cross-validation, ensuring reproducibility.

Is there specific code for updating JSON within the UNET pipeline? If so, could you kindly point it out? If not, would you be able to provide your operational code? Thank you.

**Comment by Tkrismile** [comment]
I really appreciate your work. Could you please provide some specific details about the five-fold cross-validation? For instance, does your five-fold cross-validation divide the training set into five equal parts, using four parts for training and one part for validation? Is the remaining part（validataion list in dataset.json） used as the test set?

**Comment by FengheTan9** [comment]
> I really appreciate your work. Could you please provide some specific details about the five-fold cross-validation? For instance, does your five-fold cross-validation divide the training set into five equal parts, using four parts for training and one part for validation? Is the remaining part（validataion list in dataset.json） used as the test set?

Same as UNETR, only train & validation.
Fold 1 using img0001-img0006 for validation, other cases for training.
Fold 2 using img0007-img0010, img0021, img0022 for validation, other cases for training.
Fold 3 using img0023-img0028 for validation, other cases for training.
...
Fold 5 is same as [official split](https://developer.download.nvidia.com/assets/Clara/monai/tutorials/swin_unetr_btcv_dataset_0.json).

**Comment by Tkrismile** [comment]
Thank you for your detailed explanation. I have one more question: the UNTER training framework only provides training and testing code for the BTCV dataset. Did you use the same code for training and testing on KiTS'23? How did you split the training and validation sets for the KiTS'23 dataset?
